csrc/dynamic_transform.cpp

```diff
@@ -656,7 +754,9 @@ bool DynamicTransformConcretizer::propagateFromProducerToConsumer(
         consumer->toString());
 
     auto concretized_id =
-        IterDomainBuilder(root_id).iter_type(*id_type).build();
+        IterDomainBuilder(maybeMutated(root_id)->as<IterDomain>())
```

Comment:
Is this change because the extent of `root_id` may be changed to 0?

Reply:
Yes exactly, when we call `OptOutMutator::mutate(root_id)` it will register it for mutation if any of its members are mutated, including the extent. Since we are going to register another mutation here, we don't want to lose those changes, so we base `concretized_id` on the mutated ID.